webclient/src/websocket/commands/SessionCommands.ts

```diff
@@ -110,6 +115,31 @@ export class SessionCommands {
     });
   }
 
+  static requestPasswordSalt(): void {
+    const options = webClient.options as unknown as RequestPasswordSaltParams;
+
+    const registerConfig = {
+      ...webClient.clientConfig,
+      userName: options.user,
+    };
+
+    const CmdRequestPasswordSalt = webClient.protobuf.controller.Command_RequestPasswordSalt.create(registerConfig);
+
+    const sc = webClient.protobuf.controller.SessionCommand.create({
+      ".Command_RequestPasswordSalt.ext" : CmdRequestPasswordSalt
+    });
+
+    webClient.protobuf.sendSessionCommand(sc, raw => {
+      if (raw.responseCode === webClient.protobuf.controller.Response.ResponseCode.RespOk) {
+        const passwordSalt = raw[".Response_PasswordSalt.ext"].passwordSalt;
+        SessionCommands.login(passwordSalt);
+      } else if (raw.responseCode === webClient.protobuf.controller.Response.ResponseCode.RespRegistrationRequired) {
+          SessionCommands.updateStatus(StatusEnum.DISCONNECTED, "Login failed: Registration Required");
+          SessionCommands.disconnect();
+      }
```

Comment:
Let's do a final else block that just calls disconnect, just to harden against future changes.